internal/cmd/measurement_schema/client.go

```diff
@@ -0,0 +1,284 @@
+package measurement_schema
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/influxdata/influx-cli/v2/internal"
+	"github.com/influxdata/influx-cli/v2/internal/api"
+)
+
+type Client struct {
+	BucketApi        api.BucketsApi
+	BucketSchemasApi api.BucketSchemasApi
+	CLI              *internal.CLI
+}
+
+type orgBucketID struct {
+	OrgID    string
+	BucketID string
+}
+
+func (c Client) resolveMeasurement(ctx context.Context, ids orgBucketID, name string) (string, error) {
+	res, _, err := c.BucketSchemasApi.
+		GetMeasurementSchemas(ctx, ids.BucketID).
+		OrgID(ids.OrgID).
+		Name(name).
+		Execute()
+	if err != nil {
+		return "", fmt.Errorf("failed to find measurement schema: %w", err)
+	}
+
+	if len(res.MeasurementSchemas) == 0 {
+		return "", fmt.Errorf("measurement schema %q not found", name)
+	}
+
+	return res.MeasurementSchemas[0].Id, nil
+}
+
+func (c Client) resolveOrgBucketIds(ctx context.Context, params internal.OrgBucketParams) (*orgBucketID, error) {
+	if params.OrgID.Valid() && params.BucketID.Valid() {
+		return &orgBucketID{OrgID: params.OrgID.String(), BucketID: params.BucketID.String()}, nil
+	}
+
+	if params.BucketName == "" {
+		return nil, errors.New("bucket missing: specify bucket ID or bucket name")
+	}
+
+	if !params.OrgID.Valid() && params.OrgName == "" && c.CLI.ActiveConfig.Org == "" {
+		return nil, errors.New("org missing: specify org ID or org name")
+	}
+
+	req := c.BucketApi.GetBuckets(ctx).Name(params.BucketName)
+	if params.OrgID.Valid() {
+		req = req.OrgID(params.OrgID.String())
+	} else if params.OrgName != "" {
+		req = req.Org(params.OrgName)
+	} else {
+		req = req.Org(c.CLI.ActiveConfig.Org)
+	}
+
+	resp, _, err := req.Execute()
+	if err != nil {
+		return nil, fmt.Errorf("failed to find bucket %q: %w", params.BucketName, err)
+	}
+	buckets := resp.GetBuckets()
+	if len(buckets) == 0 {
+		return nil, fmt.Errorf("no %q not found", params.BucketName)
+	}
+
+	return &orgBucketID{OrgID: buckets[0].GetOrgID(), BucketID: buckets[0].GetId()}, nil
+}
+
+func (c Client) readColumns(stdin io.Reader, f ColumnsFormat, path string) ([]api.MeasurementSchemaColumn, error) {
+	var (
+		r    io.Reader
+		name string
+	)
+
+	if path == "" {
+		r = stdin
+		name = "stdin"
+	} else {
+		data, err := os.ReadFile(path)
+		if err != nil {
+			return nil, fmt.Errorf("unable to read file %q: %w", path, err)
+		}
+		r = bytes.NewReader(data)
+		name = path
+	}
+
+	reader, err := f.DecoderFn(name)
+	if err != nil {
+		return nil, err
+	}
+
+	return reader(r)
+}
+
+type CreateParams struct {
+	internal.OrgBucketParams
+	Name           string
+	Stdin          io.Reader
+	ColumnsFile    string
+	ColumnsFormat  ColumnsFormat
+	ExtendedOutput bool
+}
+
+func (c Client) Create(ctx context.Context, params CreateParams) error {
```

Comment:
Maybe overkill, but when thinking about file layout I'd considered splitting up the main entry-points for subcommands into their own files, i.e. having a `create.go`, `update.go`, etc. WDYT?

Reply:
I don't think it is necessary at this point, however, if the commands (`List`, `Create`, etc) were to have their own distinct state and inputs, it might make sense to separate them. I'll rename the package to `bucket_schema` to match the sub-command name.